src/components/shoppingList.js

```diff
@@ -0,0 +1,53 @@
+import { useRef, useState } from 'react';
+import { AbdsBadge, AbdsButton } from '@abds/react-bindings';
+
+export const ShoppingList = () => {
+  const [itemCount, setItemCount] = useState(0);
+
+  const shoppingListRef = useRef(null);
+
+  const addItem = () => {
+    const item = document.createElement('li');
```

Comment:
this could be done without interacting with the dom directly. Just maintain an array of items with their status and text and render that out in the ul. That way no need for useRef either.